### PR TITLE
Updated the marqo.py and index.py to support marqo 2.11 ver

### PR DESCRIPTION
**Pull request by SagarSalgarr**

### CODE REVIEW

File: requirements-prod.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 transformers==4.36.0
 starlette==0.27.0
 setuptools==65.5.1
-numpy==1.26.4
+numpy
```
Comment (manojLondhe) [inline requirements-prod.txt]: add a version

File: vectorstores/marqo.py [hunk 2]
```diff
@@ -22,15 +22,13 @@ def __init__(self):
...
-            "index_defaults": {
-                "treat_urls_and_pointers_as_images": False,
-                "model": self.embedding_model,
-                "normalize_embeddings": True,
-                "text_preprocessing": {
-                    "split_length": self.SPLIT_LENGTH,
-                    "split_overlap": self.SPLIT_OVERLAP,
-                    "split_method": "sentence"
-                }
+            "treatUrlsAndPointersAsImages": False,
```
Comment (manojLondhe) [inline vectorstores/marqo.py:22]: `index_defaults`  not needed?

File: Dockerfile [hunk 3]
```diff
@@ -7,7 +7,7 @@ RUN python -m pip install --upgrade pip
 RUN pip install --no-cache-dir --upgrade -r /code/requirements-prod.txt
 COPY . /code
 
-CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8000"]
+CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8080"]
```
Comment (manojLondhe) [inline Dockerfile:10]: why port change is required?

File: index_documents.py [hunk 4]
```diff
@@ -32,50 +32,49 @@ def split_documents(documents: List[Document], chunk_size: int = 4000, chunk_ove
     """
     text_splitter = RecursiveCharacterTextSplitter(
         chunk_size=chunk_size, chunk_overlap=chunk_overlap)
-    # splited_docs = text_splitter.split_documents(documents)
-    splited_docs = []
+    pattern = re.compile(r'[\x00-\x1F\x7F\u00A0]')
```
Comment (manojLondhe) [inline index_documents.py:40]: add a comment about `what does this regex look for`

File: vectorstores/marqo.py [hunk 5]
```diff
@@ -84,6 +82,27 @@ def add_documents(self, documents=List[Document], fresh_collection: bool = False
...
-        documents = docsearch.similarity_search_with_score(query, k)
-        return documents
+        try:
+            docsearch = Marqo(self.client, index_name=collection_name)
+            documents = docsearch.similarity_search_with_score(query, k)
+            return documents
+        except Exception as e:
+            return []
+    
+    def cache_documents(self, documents: List[Dict[str, str]], collection_name:str) -> str:
```
Comment (manojLondhe) [inline vectorstores/marqo.py]: add method level comment about 
`cache_documents`

File: Dockerfile (hunk 3 above)
Comment (SagarSalgarr) [reply Dockerfile:10]: In production sakhi api service is running on 8080 , so we need to change the port  in dockerfile.

File: vectorstores/marqo.py (hunk 2 above)
Comment (SagarSalgarr) [reply vectorstores/marqo.py:22]: yes. this is the main change (or we can say upgrade)

File: requirements-prod.txt [hunk 6]
```diff
@@ -1,7 +1,7 @@
 transformers==4.36.0
 starlette==0.27.0
 setuptools==65.5.1
-numpy==1.26.4
+numpy==1.22.2
```
Comment (manojLondhe) [inline requirements-prod.txt:4]: why are we downgrading?

File: index_documents.py [hunk 7]
```diff
@@ -33,13 +34,17 @@ def split_documents(documents: List[Document], chunk_size: int = 4000, chunk_ove
...
     splited_docs = []
     for document in documents:
         for chunk in text_splitter.split_text(document.text):
+            chunk = re.sub(pattern, '', chunk)
             splited_docs.append(Document(page_content=chunk, metadata={
                 "page_label": document.metadata.get("page_label"),
                 "file_name": document.metadata.get("file_name"),
-                "file_path": document.metadata.get("file_path"),
+                # "file_path": document.metadata.get("file_path"),
```
Comment (manojLondhe) [inline index_documents.py]: not needed anymore?

File: requirements-prod.txt (hunk 6 above)
Comment (SagarSalgarr) [reply requirements-prod.txt:4]: Previous version  was throwing error while doing indexing on production so we downgraded the numpy version

File: index_documents.py (hunk 7 above)
Comment (SagarSalgarr) [reply index_documents.py]: As we are taking file path from command line argument while indexing so i commented it out as we don't need it anymore

File: index_documents.py [hunk 8]
```diff
@@ -1,12 +1,12 @@
...
-    List
-)
+import re
+from typing import List
 from langchain.docstore.document import Document
 from langchain.text_splitter import RecursiveCharacterTextSplitter
 from llama_index import SimpleDirectoryReader
-from env_manager import vectorstore_class
- 
+from vectorstores.marqo import MarqoVectorStore
```
Comment (manojLondhe) [inline index_documents.py]: we may revert this and do it in generalised way

File: index_documents.py [hunk 9]
```diff
@@ -69,26 +68,33 @@ def indexer_main():
...
     CHUNK_SIZE = args.chunk_size
     CHUNK_OVERLAP = args.chunk_overlap
+    INDEX_NAME = args.index_name
 
     documents = load_documents(FOLDER_PATH, CHUNK_SIZE, CHUNK_OVERLAP)
     print("Total documents :: =>", len(documents))
     
     print("Adding documents...")
-    results = vectorstore_class.add_documents(documents, FRESH_INDEX)
+    vector_store = MarqoVectorStore(index_name=INDEX_NAME)
```
Comment (manojLondhe) [inline index_documents.py]: we may revert this and do it in generalised way

**Pull request merged**